sklearn/linear_model/_linear_loss.py

```diff
@@ -287,13 +287,29 @@ def loss_gradient(
 
         if not self.base_loss.is_multiclass:
             grad = np.empty_like(coef, dtype=weights.dtype)
-            grad[:n_features] = X.T @ grad_pointwise + l2_reg_strength * weights
+            with np.errstate(all="raise"):
+                try:
+                    grad[:n_features] = X.T @ grad_pointwise + l2_reg_strength * weights
+                except FloatingPointError:
```

Comment:
```suggestion
                except FloatingPointError as exc:
```

Reply:
When I use ```exc``` as you have in this suggestion, testing on the dataset from I get the following:

```
---------------------------------------------------------------------------
FloatingPointError                        Traceback (most recent call last)
File [~/Downloads/scikit-learn/sklearn/linear_model/_linear_loss.py:296](https://file+.vscode-resource.vscode-cdn.net/Users/d0055315_akaash_patel/Downloads/scikit-learn/scratchpad~/~/Downloads/scikit-learn/sklearn/linear_model/_linear_loss.py:296), in LinearModelLoss.loss_gradient(self, coef, X, y, sample_weight, l2_reg_strength, n_threads, raw_prediction)
    [295](https://file+.vscode-resource.vscode-cdn.net/Users/d0055315_akaash_patel/Downloads/scikit-learn/scratchpad~/~/Downloads/scikit-learn/sklearn/linear_model/_linear_loss.py:295) try:
--> [296](https://file+.vscode-resource.vscode-cdn.net/Users/d0055315_akaash_patel/Downloads/scikit-learn/scratchpad~/~/Downloads/scikit-learn/sklearn/linear_model/_linear_loss.py:296)     grad[:n_features] = X.T @ grad_pointwise + l2_reg_strength * weights
    [297](https://file+.vscode-resource.vscode-cdn.net/Users/d0055315_akaash_patel/Downloads/scikit-learn/scratchpad~/~/Downloads/scikit-learn/sklearn/linear_model/_linear_loss.py:297)     print("min entry", np.min(grad_pointwise))

FloatingPointError: invalid value encountered in matmul

The above exception was the direct cause of the following exception:```

so it still includes the original exception that I am "wrapping" with a ValueError and message. Do you reckon it's better this way?